department-of-veterans-affairs/caseflow
# Mkerns/caseflow 859 nod form validation

Resolves [CASEFLOW-859](https://vajira.max.gov/browse/CASEFLOW-859)

### Description
Removed use of MomentJS in replace of YupJS Validation package. This allows for better control of form submissions with various situational validations.

### Acceptance Criteria
- [ ] Refactor edit NOD from validation (to replace existing date validation logic as well as reason validation) to use Yup validation. 
- [ ] Should also resolve being able to submit an incomplete date.

### Testing Plan
1. Select user with access to Edit NOD link e.g. BVAISHAW (VACO)
2. Go to the case details page of an appeal
3. Click the "Edit NOD Date" Button that appears under Case Timeline section
4. Go to Queue and search for a veteran id like '701305078'
5. Enter date before 01-01-2018
6. Verify an NOD Date error message appears and submit button does not submit the form
7. Next, enter date after today's date
8. Verify an NOD Date error message appears and submit button does not submit the form
9. Next, enter 02/21/yyyy
10. Verify an NOD Date error message appears and submit button does not submit the form
11. Next, enter a valid date that is after the original NOD Date
12. Verify that the warning modal appears
13. Then click submit without selecting a reason
14. Verify a reason error message appears and submit button does not submit the form
15. Next select a reason and click submit
16. Verify the form submits and the success banner appears.
